plaso/parsers/plist_plugins/safari_downloads.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+"""Plist parser plugin for Safari Downloads.plist files."""
+
+from plaso.containers import events
+from plaso.parsers import plist
+from plaso.parsers.plist_plugins import interface
+
+
+class SafariDownloadEventData(events.EventData):
+  """Safari download event data.
+
+  Attributes:
+    end_time (dfdatetime.DateTimeValues): date and time the download was
+        finished.
+    full_path (str): full path where the file was downloaded to.
+    received_bytes (int): number of bytes received while downloading.
+    start_time (dfdatetime.DateTimeValues): date and time the download was
+        started.
+    total_bytes (int): total number of bytes to download.
+    url (str): URL of the downloaded file.
+  """
+
+  DATA_TYPE = 'safari:downloads:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(SafariDownloadEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.end_time = None
+    self.full_path = None
+    self.received_bytes = None
+    self.start_time = None
+    self.total_bytes = None
+    self.url = None
```

Comment:
Just the field I had, except I also added `removed_on_complete := event_data.DownloadEntryRemoveWhenDoneKey`

Reply:
https://blog.d204n6.com/2021/05/ios-macos-tracking-downloads-from.html hints that this value is used for "remove on close"

Reply:
I've added it, update pending shortly